terraform/modules/happy-ingress-eks/main.tf

```diff
@@ -1,52 +1,114 @@
-    "alb.ingress.kubernetes.io/listen-ports" = jsonencode(local.listen_ports_tls)
+    "alb.ingress.kubernetes.io/listen-ports"         = jsonencode([{ HTTPS = 443 }, { HTTP = 80 }])
     # All ingresses are "internet-facing". If a service_type was marked "INTERNAL", it will be protected using OIDC.
     "alb.ingress.kubernetes.io/scheme"                  = "internet-facing"
     "alb.ingress.kubernetes.io/subnets"                 = join(",", var.cloud_env.public_subnets)
     "alb.ingress.kubernetes.io/success-codes"           = var.routing.success_codes
     "alb.ingress.kubernetes.io/tags"                    = var.tags_string
     "alb.ingress.kubernetes.io/target-group-attributes" = "deregistration_delay.timeout_seconds=60"
     "alb.ingress.kubernetes.io/target-type"             = "instance"
-    "kubernetes.io/ingress.class"                       = "alb"
     "alb.ingress.kubernetes.io/group.name"              = var.routing.group_name
     "alb.ingress.kubernetes.io/group.order"             = var.routing.priority
   }
+
+  redirect_action_name = "redirect"
   ingress_tls_annotations = {
-    "alb.ingress.kubernetes.io/actions.redirect" = <<EOT
-        {"Type": "redirect", "RedirectConfig": {"Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}
-      EOT
-    "alb.ingress.kubernetes.io/certificate-arn"  = var.certificate_arn
-    "alb.ingress.kubernetes.io/ssl-policy"       = "ELBSecurityPolicy-TLS-1-2-2017-01"
+    "alb.ingress.kubernetes.io/actions.${local.redirect_action_name}" = jsonencode({
+      Type = local.redirect_action_name
+      RedirectConfig = {
+        Protocol   = "HTTPS"
+        Port       = "443"
+        StatusCode = "HTTP_301"
+      }
+    })
+    "alb.ingress.kubernetes.io/certificate-arn" = var.certificate_arn
+    "alb.ingress.kubernetes.io/ssl-policy"      = "ELBSecurityPolicy-TLS-1-2-2017-01"
   }
+
   ingress_auth_annotations = {
     "alb.ingress.kubernetes.io/auth-type"                       = "oidc"
     "alb.ingress.kubernetes.io/auth-on-unauthenticated-request" = "authenticate"
     "alb.ingress.kubernetes.io/auth-idp-oidc"                   = jsonencode(var.routing.oidc_config)
   }
+
   ingress_annotations = (
     var.routing.service_type == "EXTERNAL" ?
     merge(local.ingress_tls_annotations, local.ingress_base_annotations) :
     merge(local.ingress_tls_annotations, local.ingress_auth_annotations, local.ingress_base_annotations)
   )
+
+  // the length of bypasses should never be bigger than the priority due to how we do the spread priority
+  // in happy-stack-eks. We also have a validation on the input variable to ensure this.
+  priority_spread = range(var.routing.priority - length(var.routing.bypasses), var.routing.priority)
+  routing_keys    = keys(var.routing.bypasses)
+  ingress_bypass_annotations = ({
+    for k, v in zipmap(local.routing_keys, local.priority_spread) : k =>
+    merge(
+      local.ingress_base_annotations,
```

Comment:
`var.routing.priority` is the base routing number and the priority_spread is the set of overrides to use. Here `v` is assigned a number smaller than `var.routing.priority`